### PR TITLE
Fix optimized everyone sleeping breaking spawnpoints

### PR DESCRIPTION
**Pull request by Tigermouthbear**
this fixes spawnpoints not saving

### CODE REVIEW

**Comment by 5HT2** [comment]
so this is why my bed didn't work

File: src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt [hunk 1]
```diff
@@ -24,7 +24,11 @@ object OptimisedEveryoneSleeping: PaperFeature {
 					method.instructions.clear()
 					method.instructions.add(insnBuilder {
 						+VarInsnNode(ALOAD, 0)
-						+MethodInsnNode(INVOKESTATIC, OptimisedEveryoneSleeping::class.internalName, "everyoneDeeplySleeping", "(Ljava/lang/Object;)Z", false)
+						+VarInsnNode(ALOAD, 0) // load Q
+
+						+VarInsnNode(ALOAD, 0) // load `this`
```
Comment (fee1-dead) [inline src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt:31]: Why are there 3 ALOADs?

Comment (fee1-dead) [reply src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt:31]: And why did you aload `this` two times?

File: src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt [hunk 2]
```diff
@@ -40,12 +44,12 @@ object OptimisedEveryoneSleeping: PaperFeature {
...
 	@JvmStatic
-	fun everyoneDeeplySleeping(worldServer: Any): Boolean {
+	fun everyoneDeeplySleeping(flag: Boolean, worldServer: Any): Boolean {
 		worldServer as WorldServer
 		
 		val players = worldServer.players
-		
-		return if (players.size == 0)
+
+		return if (players.size == 0 || worldServer.isClientSide || !flag)
```
Comment (fee1-dead) [inline src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt:54]: When is `isClientSide` true?
Isn't this a server?

**Review by fee1-dead — CHANGES_REQUESTED** [review]

File: src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt (hunk 2 above)
Comment (Tigermouthbear) [reply src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt:54]: idk, mc code checks that

File: src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt (hunk 1 above)
Comment (Tigermouthbear) [reply src/main/kotlin/dev/binclub/paperbin/transformers/OptimisedEveryoneSleeping.kt:31]: bc im dumb

**Comment by x4e** [comment]
Thanks :)

**Pull request merged**